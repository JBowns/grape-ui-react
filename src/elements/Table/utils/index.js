export * from './constants';
export * from './cssDefaults';
export * from './props';
export * from './styled';
export * from './viewState';
