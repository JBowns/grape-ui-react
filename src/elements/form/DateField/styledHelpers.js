import { css } from 'styled-components';
import { variant } from 'styled-system';

export const menuStyledHelper = css`
  .DayPickerInput-OverlayWrapper {
    pointer-events: none;
    ${variant({
    prop: 'menuAlignment',
    variants: {
      left: {
        left: '1rem',
        right: 'auto',
      },
      right: {
        left: 'auto',
        right: '-1rem',
      },
    },
  })}
    .DayPickerInput-Overlay {
      justify-content: flex-end;
      width: 100%;
      ${variant({
    prop: 'menuAlignment',
    variants: {
      left: {
        display: 'block',
      },
      right: {
        display: 'flex',
      },
    },
  })}
    }
  }
  .DayPicker-Months {
    flex-wrap: nowrap;
    ${variant({
    prop: 'menuDirection',
    variants: {
      column: {
        flexDirection: 'column',
      },
      row: {
        flexDirection: 'row',
        minWidth: 625,
      },
    },
  })}
  }
`;
