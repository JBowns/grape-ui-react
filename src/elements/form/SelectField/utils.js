import {
  defaultControlStyles,
  resolveBoxShadow,
  resolveColor,
  resolveZIndex,
} from 'src/utils/styledHelpers';
import { getGlobalOverrides } from 'src/global-styles';

export const styleOverridesBase = ({ ...props }) => {
  const {
    menuElevation,
    menuZIndex,
    styleOverrides,
  } = props;
  const globalOverrides = getGlobalOverrides(props);
  const multiValueMargin = 2;
  const styleZIndex = menuZIndex || resolveZIndex(menuElevation, globalOverrides);
  const resolveBackground = ({ isFocused, isSelected }) => {
    let background = 'inherit';
    if (isFocused) {
      background = resolveColor('brandLinkHover', globalOverrides);
    } else if (isSelected) {
      background = resolveColor('brandLink', globalOverrides);
    }
    return background;
  };
  return {
    clearIndicator: provided => ({
      ...provided,
      padding: '0',
      ...styleOverrides.clearIndicator,
    }),
    control: () => ({
      border: '0',
      boxShadow: 'none',
      display: 'flex',
      minHeight: '0',
      ...styleOverrides.control,
    }),
    dropdownIndicator: provided => ({
      ...provided,
      padding: '0',
      ...styleOverrides.dropdownIndicator,
    }),
    indicatorsContainer: provided => ({
      ...provided,
      bottom: '0',
      justifyContent: 'flex-end',
      padding: '0',
      ...styleOverrides.indicatorsContainer,
    }),
    indicatorSeparator: () => ({
      display: 'none',
      ...styleOverrides.indicatorSeparator,
    }),
    input: styles => null,
    menu: provided => ({
      ...provided,
      boxShadow: resolveBoxShadow(menuElevation, globalOverrides),
      left: '0',
      marginBottom: '0',
      marginTop: '2px',
      padding: '8px 0',
      position: 'absolute',
      width: '100%',
      zIndex: styleZIndex,
      ...styleOverrides.menu,
    }),
    menuPortal: provided => ({
      ...provided,
      zIndex: styleZIndex,
      ...styleOverrides.menuPortal,
    }),
    multiValue: () => ({
      backgroundColor: 'rgba(0, 0, 0, 0.1)',
      borderRadius: '4px',
      display: 'flex',
      fontSize: '80%',
      margin: multiValueMargin,
      padding: '4px',
      ...styleOverrides.multiValue,
    }),
    multiValueLabel: () => ({
      padding: '0 4px',
      ...styleOverrides.multiValueLabel,
    }),
    multiValueRemove: () => ({
      alignItems: 'center',
      cursor: 'pointer',
      display: 'flex',
      ...styleOverrides.multiValueRemove,
    }),
    option: (styles, { isFocused, isSelected }) => ({
      background: resolveBackground({ isFocused, isSelected }),
      color: isFocused || isSelected ? resolveColor('white', globalOverrides) : 'inherit',
      padding: '8px 16px',
      ...styleOverrides.option,
    }),
    placeholder: () => ({
      color: resolveColor(defaultControlStyles.placeholderColor, globalOverrides),
      ...styleOverrides.placeholder,
    }),
    singleValue: () => ({
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      whiteSpace: 'nowrap',
      ...styleOverrides.singleValue,
    }),
    valueContainer: (provided, { hasValue, isMulti }) => ({
      ...provided,
      flex: '1 1 0',
      flexWrap: isMulti ? 'wrap' : 'nowrap',
      margin: hasValue && isMulti ? multiValueMargin * -1.5 : null,
      padding: '0',
      ...styleOverrides.valueContainer,
    }),
  };
};
