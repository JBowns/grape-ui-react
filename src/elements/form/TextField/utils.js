const cleaveMap = new Map();
cleaveMap.set('phone', {
  delimiter: '-',
  phone: true,
  phoneRegionCode: 'US',
});
cleaveMap.set('currency', {
  numeral: true,
  numeralThousandsGroupStyle: 'thousand',
  prefix: '$',
});
cleaveMap.set('integer', {
  numeral: true,
  numeralDecimalScale: 0,
  numeralThousandsGroupStyle: 'thousand',
});
cleaveMap.set('numeric', {
  numeral: true,
  numeralThousandsGroupStyle: 'thousand',
});
cleaveMap.set('postalCode', {
  blocks: [5, 4],
  delimiter: '-',
  delimiterLazyShow: true,
  numericOnly: true,
});

export const isCleaveInput = props => {
  if (props.email || props.password) {
    return false;
  }
  if (props.formatterOptions) {
    return true;
  }
  let foundKey = false;
  const checkKeys = (value, key) => {
    if (props[key] && !foundKey) {
      foundKey = true;
    }
  };

  cleaveMap.forEach(checkKeys);
  return foundKey;
};

export const cleaveOption = props => {
  if (props.formatterOptions) {
    return props.formatterOptions;
  }
  let options = null;
  const setOptionWhenFound = (value, key) => {
    if (props[key]) {
      options = value;
    }
  };

  cleaveMap.forEach(setOptionWhenFound);
  return options;
};
