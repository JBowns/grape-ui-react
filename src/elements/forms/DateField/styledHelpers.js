import { css } from 'styled-components';
import { system, variant } from 'styled-system';

export const menuStyledHelper = props => css`
  .DayPickerInput-OverlayWrapper {
    pointer-events: none;
    .DayPickerInput-Overlay {
  ${variant({
    prop: 'menuPlacement',
    variants: {
      bottom: {},
      top: {
        transform: `translateX(0) translateY(calc(-100% - ${props.controlHeight}))`,
      },
    },
  })}
    }
  ${variant({
    prop: 'menuAlignment',
    variants: {
      left: {
        left: '1rem',
        right: 'auto',
      },
      right: {
        left: 'auto',
        right: '-1rem',
      },
    },
  })}
    ${system({
    menuOverlayBottom: { property: 'bottom' },
    menuOverlayLeft: { property: 'left' },
    menuOverlayRight: { property: 'right' },
    menuOverlayTop: { property: 'top' },
  })}

    .DayPickerInput-Overlay {
      justify-content: flex-end;
      width: 100%;
  ${variant({
    prop: 'menuAlignment',
    variants: {
      left: {
        display: 'block',
      },
      right: {
        display: 'flex',
      },
    },
  })}
    }
  }
  .DayPicker-Months {
    flex-wrap: nowrap;
  ${variant({
    prop: 'menuDirection',
    variants: {
      column: {
        flexDirection: 'column',
      },
      row: {
        flexDirection: 'row',
        minWidth: 625,
      },
    },
  })}
  }
`;
