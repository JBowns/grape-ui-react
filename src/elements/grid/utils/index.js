import propTypes from '@styled-system/prop-types';

export const defaultFlexBoxStylesPropsToTrim = [
  ...Object.keys(propTypes.background),
  ...Object.keys(propTypes.border),
  ...Object.keys(propTypes.flexbox),
  ...Object.keys(propTypes.grid),
  ...Object.keys(propTypes.layout),
  ...Object.keys(propTypes.position),
  ...Object.keys(propTypes.shadow),
  ...Object.keys(propTypes.space),
  'boxSizing',
];
