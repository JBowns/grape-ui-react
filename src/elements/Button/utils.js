import { style } from 'styled-system';
import { POSITION_DEFAULT_VALUE, resolveColor } from 'src/utils/styledHelpers';
import { DEFAULT_BUTTON_LINE_HEIGHT } from './constants';

export const lineHeightStyle = style({
  // The corresponding CSS property (defaults to prop argument)
  cssProperty: 'line-height',
  // key for theme values
  key: 'lineHeight',
  // React prop name
  prop: 'lineHeight',
  // add a fallback scale object or array, if theme is not present
  scale: ['', '', '', '', ''],
  // accessor function for transforming the value
  transformValue: lineHeight => parseFloat(lineHeight) || DEFAULT_BUTTON_LINE_HEIGHT,
});

export const lineHeightButton = props => (props.lineHeight ? lineHeightStyle(props) : `line-height: ${DEFAULT_BUTTON_LINE_HEIGHT};`);

const positionStyle = style({
  // The corresponding CSS property (defaults to prop argument)
  cssProperty: 'position',
  // key for theme values
  key: 'position',
  // React prop name
  prop: 'position',
  // add a fallback scale object or array, if theme is not present
  scale: ['', '', '', '', ''],
  // accessor function for transforming the value
  transformValue: position => position || POSITION_DEFAULT_VALUE,
});

export const positionButton = props => (props.position ? positionStyle(props) : `position: ${POSITION_DEFAULT_VALUE};`);

const hasVariant = variant => {
  if (!variant) {
    return false;
  }
  return !!variant;
};

export const activeColorButton = props => {
  const { bgActiveColor, variant } = props;
  if (hasVariant(variant)) {
    return null;
  }
  const color = (!bgActiveColor || bgActiveColor === null) ? resolveColor('white.light') : resolveColor(bgActiveColor);
  return `background-color: ${color}`;
};

export const hoverColorButton = props => {
  const { bgHoverColor, variant } = props;
  if (hasVariant(variant)) {
    return null;
  }
  const color = (!bgHoverColor || bgHoverColor === null) ? resolveColor('white.dark') : resolveColor(bgHoverColor);
  return `background-color: ${color}`;
};
