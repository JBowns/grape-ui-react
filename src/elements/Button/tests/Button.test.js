import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { getGlobalStyles, getGlobalOverrides } from 'src/global-styles';
import { MemoryRouter as Router } from 'react-router-dom';
import { Button } from '..';
import { buttonThemes } from 'src/utils/styledHelpers';

const assertReactElement = element => {
  const component = renderer.create(element);
  return component.toJSON();
};
describe('Button base', () => {
  it('should return a Button', () => {
    const element = (
      <Button>Button Base</Button>
    );
    expect(assertReactElement(element)).toMatchSnapshot();
  });

  it('should return an outlined Button', () => {
    const element = <Button id="buttonOutline" outline>Button Outline</Button>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });

  it('should return an contained Button', () => {
    const element = <Button contained id="buttonContained">Button Contained</Button>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });

  it('should return an contained raised Button', () => {
    const element = <Button contained id="buttonContainedRaised" raised>Button Contained and Raised</Button>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });

  it('should return a Button with disabled', () => {
    const element = <Button disabled id="exampleColor">Disabled button</Button>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
});

describe('Button Component base with style', () => {
  it('should return a Button with bg and color', () => {
    const element = <Button bg="green" color="white" id="exampleColor">Happy St. Patrick&#39;s Day!</Button>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Button with bg and color and bgActiveColor and bgHoverColor ', () => {
    const element = <Button bg="green" bgActiveColor="green.light" bgHoverColor="green.dark" color="white" id="exampleColor">Happy St. Patrick&#39;s Day!</Button>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Button with sm', () => {
    const element = <Button id="exampleColor" sm>Do this</Button>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Button with lg', () => {
    const element = <Button id="exampleColor" lg>Do this</Button>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });

  it('should return a Button with a zIndex when using a string', () => {
    const element = <Button zIndex="500">Do this</Button>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Button with a zIndex when using a number', () => {
    const element = <Button zIndex={500}>Do this</Button>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Link with href with button look', () => {
    const element = <Button href="https://www.google.com/" target="_blank">google</Button>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return a Link with to with button look', () => {
    const element = <Router><Button to="/404">404</Button></Router>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
});

describe('Button using theme colors', () => {
  it('should return a Button', () => {
    const globalStyles = getGlobalStyles();
    const { colors } = globalStyles;
    const theme = {
      colors: {
        ...colors,
        yellow: {
          base: 'hsl(65, 65%, 50%)',
          dark: 'hsl(65, 65%, 25%)',
          light: 'hsl(65, 65%, 68%)',
        },
      },
      space: [0, 4, 8, 16],
    };
    const element = (
      <ThemeProvider theme={theme}>
        <Button bg="yellow" color="yellow.dark">Do this</Button>
      </ThemeProvider>
    );
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  describe('Button using styled system fontSize', () => {
    it('should return a Button with specified font size', () => {
      const element = (
        <Button fontSize={3} variant="primary">Do this</Button>
      );
      expect(assertReactElement(element)).toMatchSnapshot();
    });
  });
});

describe('Button using variant styles', () => {
  const theme = {
    buttons: {
      ...buttonThemes(),
    },
    space: [0, 4, 8, 16],
  };
  describe('Button with variant set to primary', () => {
    it('should return a Button with primary styling', () => {
      const element = (
        <ThemeProvider theme={theme}>
          <Button variant="primary">Do this</Button>
        </ThemeProvider>
      );
      expect(assertReactElement(element)).toMatchSnapshot();
    });
  });
  describe('Button with variant set to danger', () => {
    it('should return a Button with danger styling', () => {
      const element = (
        <ThemeProvider theme={theme}>
          <Button variant="danger">Do this</Button>
        </ThemeProvider>
      );
      expect(assertReactElement(element)).toMatchSnapshot();
    });
  });
  describe('Button with variant set to contained-light', () => {
    it('should return a Button with contained-light styling', () => {
      const element = (
        <ThemeProvider theme={theme}>
          <Button variant="contained-light">Do this</Button>
        </ThemeProvider>
      );
      expect(assertReactElement(element)).toMatchSnapshot();
    });
  });
  describe('Button with variant set to contained-danger', () => {
    it('should return a Button with contained-danger styling', () => {
      const element = (
        <ThemeProvider theme={theme}>
          <Button variant="contained-danger">Do this</Button>
        </ThemeProvider>
      );
      expect(assertReactElement(element)).toMatchSnapshot();
    });
  });
  describe('Button set to an unknown variant', () => {
    it('should return a Button with no variant styling', () => {
      const element = (
        <ThemeProvider theme={theme}>
          <Button variant="not-a-variant">Do this</Button>
        </ThemeProvider>
      );
      expect(assertReactElement(element)).toMatchSnapshot();
    });
  });
  describe('Button set with no variant', () => {
    it('should return a Button with bg and color styling preserved', () => {
      const element = (
        <ThemeProvider theme={theme}>
          <Button bg="yellow" color="yellow.dark">Do this</Button>
        </ThemeProvider>
      );
      expect(assertReactElement(element)).toMatchSnapshot();
    });
  });
  describe('Button set with variant, bg and color', () => {
    it('should return a Button with bg and color styling overwriting the variant bg and color style', () => {
      const element = (
        <ThemeProvider theme={theme}>
          <Button bg="yellow" color="yellow.dark" variant="contained-danger">Do this</Button>
        </ThemeProvider>
      );
      expect(assertReactElement(element)).toMatchSnapshot();
    });
  });
});

describe('Button using variant styles with theme colors', () => {
  const props = {
    theme: {
      colors: {
        brandPrimary: {
          base: 'hsl(323.31, 85.61%, 29.98%)',
          dark: 'hsl(302.91, 34.91%, 24.71%)',
          light: 'hsl(312.81, 68.81%, 48.51%)',
        },
      },
    },
  };
  const globalOverrides = getGlobalOverrides(props);
  const theme = {
    buttons: {
      ...buttonThemes(null, globalOverrides),
    },
    space: [0, 4, 8, 16],
  };
  describe('Button with variant set to contained-primary', () => {
    it('should return a Button with contained-primary styling based on theme.colors', () => {
      const element = (
        <ThemeProvider theme={theme}>
          <Button variant="contained-primary">Do this</Button>
        </ThemeProvider>
      );
      expect(assertReactElement(element)).toMatchSnapshot();
    });
  });
});
