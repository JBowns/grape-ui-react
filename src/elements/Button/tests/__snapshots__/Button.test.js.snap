// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component base with style should return a Button with a zIndex when using a number 1`] = `
.c0 {
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  display: inline-block;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  z-index: 500;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

<button
  className="c0"
  type="button"
>
  Do this
</button>
`;

exports[`Button Component base with style should return a Button with a zIndex when using a string 1`] = `
.c0 {
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  display: inline-block;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  z-index: 500;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

<button
  className="c0"
  type="button"
>
  Do this
</button>
`;

exports[`Button Component base with style should return a Button with bg and color 1`] = `
.c0 {
  border-radius: 4px;
  color: hsl(0,0%,98%);
  background-color: hsl(122.4,39.4%,49.2%);
  display: inline-block;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

<button
  className="c0"
  id="exampleColor"
  type="button"
>
  Happy St. Patrick's Day!
</button>
`;

exports[`Button Component base with style should return a Button with bg and color and bgActiveColor and bgHoverColor  1`] = `
.c0 {
  border-radius: 4px;
  color: hsl(0,0%,98%);
  background-color: hsl(122.4,39.4%,49.2%);
  display: inline-block;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(122.4,39.4%,29.2%);
}

.c0:active {
  background-color: hsl(122.4,39.4%,69.2%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

<button
  className="c0"
  id="exampleColor"
  type="button"
>
  Happy St. Patrick's Day!
</button>
`;

exports[`Button Component base with style should return a Button with lg 1`] = `
.c0 {
  border-radius: 8px;
  color: inherit;
  background-color: transparent;
  display: inline-block;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1.2rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 32px;
  padding-right: 32px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

<button
  className="c0"
  id="exampleColor"
  type="button"
>
  Do this
</button>
`;

exports[`Button Component base with style should return a Button with sm 1`] = `
.c0 {
  border-radius: 2px;
  color: inherit;
  background-color: transparent;
  display: inline-block;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 8px;
  padding-right: 8px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

<button
  className="c0"
  id="exampleColor"
  type="button"
>
  Do this
</button>
`;

exports[`Button Component base with style should return a Link with href with button look 1`] = `
.c0 {
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  display: inline-block;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

<a
  className="c0"
  href="https://www.google.com/"
  target="_blank"
>
  google
</a>
`;

exports[`Button Component base with style should return a Link with to with button look 1`] = `
.c0 {
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  display: inline-block;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

<a
  className="c0"
  href="/404"
  onClick={[Function]}
  target="_self"
>
  404
</a>
`;

exports[`Button base should return a Button 1`] = `
.c0 {
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  display: inline-block;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

<button
  className="c0"
  type="button"
>
  Button Base
</button>
`;

exports[`Button base should return a Button with disabled 1`] = `
.c0 {
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  display: inline-block;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

<button
  className="c0"
  disabled={true}
  id="exampleColor"
  type="button"
>
  Disabled button
</button>
`;

exports[`Button base should return an contained Button 1`] = `
.c0 {
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  display: inline-block;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

<button
  className="c0"
  id="buttonContained"
  type="button"
>
  Button Contained
</button>
`;

exports[`Button base should return an contained raised Button 1`] = `
.c0 {
  border-radius: 4px;
  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14),0 3px 1px -2px rgba(0,0,0,0.12),0 1px 5px 0 rgba(0,0,0,0.2);
  color: inherit;
  background-color: transparent;
  display: inline-block;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  box-shadow: 0 4px 5px 0 rgba(0,0,0,0.14),0 1px 10px 0 rgba(0,0,0,0.12),0 2px 4px -1px rgba(0,0,0,0.3);
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

<button
  className="c0"
  id="buttonContainedRaised"
  type="button"
>
  Button Contained and Raised
</button>
`;

exports[`Button base should return an outlined Button 1`] = `
.c0 {
  border: 1px solid hsl(0,0%,25%);
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  display: inline-block;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c0:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c0:active {
  background-color: hsl(0,0%,100%);
}

.c0[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

<button
  className="c0"
  id="buttonOutline"
  type="button"
>
  Button Outline
</button>
`;
